subsys/dfu/boot/mcuboot.c

```diff
@@ -151,24 +150,23 @@ int mcuboot_swap_type_multi(int image_index)
 
 int mcuboot_swap_type(void)
 {
-#ifdef FLASH_AREA_IMAGE_SECONDARY
+#if ZEPHYR_MCUBOOT_APP_0_SECONDARY_SLOT_EXISTS
 	return boot_swap_type();
 #else
 	return BOOT_SWAP_TYPE_NONE;
-#endif
-
+#endif /* !ZEPHYR_MCUBOOT_APP_0_SECONDARY_SLOT_EXISTS */
 }
 
 int boot_request_upgrade(int permanent)
 {
-#ifdef FLASH_AREA_IMAGE_SECONDARY
+#if ZEPHYR_MCUBOOT_APP_0_SECONDARY_SLOT_EXISTS
```

Comment:
You may have secondary slot,  as visible from MCUmgr, in outside flash that you set flags to but it belongs to other CPU.
Anyway, this is topic for system builds where system build configuration of MCUboot may superimpose some configuration on application it will be booting.